src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs

```diff
@@ -163,18 +176,41 @@ protected virtual void QueueBlazorStart()
 		{
 		}
 
-		/// <summary>
-		/// Override this method to manage opening links in the webview. Not all platforms require this.
-		/// </summary>
-		protected virtual void CoreWebView2_NavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs args)
+		private void CoreWebView2_NavigationStarting(object sender, CoreWebView2NavigationStartingEventArgs args)
 		{
+			if (Uri.TryCreate(args.Uri, UriKind.RelativeOrAbsolute, out var uri) && uri.Host != AppHostAddress)
+			{
+				var callbackArgs = new ExternalLinkNavigationEventArgs(uri);
+				_externalNavigationStarting?.Invoke(callbackArgs);
+
+				if (callbackArgs.ExternalLinkNavigationPolicy == ExternalLinkNavigationPolicy.OpenInExternalBrowser)
+				{
+					LaunchUri(uri);
+				}
+
+				args.Cancel = callbackArgs.ExternalLinkNavigationPolicy != ExternalLinkNavigationPolicy.OpenInWebView;
+			}
 		}
 
-		/// <summary>
-		/// Override this method to manage opening a new window in the webview. Not all platforms require this.
-		/// </summary>
-		protected virtual void CoreWebView2_NewWindowRequested(object sender, CoreWebView2NewWindowRequestedEventArgs args)
+		private void CoreWebView2_NewWindowRequested(object sender, CoreWebView2NewWindowRequestedEventArgs args)
 		{
+			// Intercept _blank target <a> tags to always open in device browser.
+			// The ExternalLinkCallback is not invoked.
+			if (Uri.TryCreate(args.Uri, UriKind.RelativeOrAbsolute, out var uri))
+			{
+				LaunchUri(uri);
+				args.Handled = true;
+			}
+		}
+
+		private void LaunchUri(Uri uri)
+		{
+#if WEBVIEW2_WINFORMS || WEBVIEW2_WPF
+			Process.Start(@"C:\Program Files\Internet Explorer\iexplore.exe", uri.ToString());
```

Comment:
Definitely can't hard-code this. It's actually even easier: Just call `Process.Start()` with the URL! Windows will Do The Right Thing:tm:

More info: https://docs.microsoft.com/troubleshoot/developer/visualstudio/csharp/general/start-internet-browser#use-the-process-class-start-method-to-start-the-browser

Reply:
Hmm I'd tried that but unfortunately it didn't do the right thing 😞 .

```
System.ComponentModel.Win32Exception
  HResult=0x80004005
  Message=An error occurred trying to start process 'https://microsoft.com/' with working directory 'C:\Users\taparik\dev\maui\src\BlazorWebView\samples\BlazorWpfApp\bin\Debug\net6.0-windows'. The system cannot find the file specified.
  Source=System.Diagnostics.Process
  StackTrace:
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName)
   at Microsoft.AspNetCore.Components.WebView.WebView2.WebView2WebViewManager.LaunchUri(Uri uri) in C:\Users\taparik\dev\maui\src\BlazorWebView\src\SharedSource\WebView2WebViewManager.cs:line 209
   at Microsoft.AspNetCore.Components.WebView.WebView2.WebView2WebViewManager.CoreWebView2_NewWindowRequested(Object sender, CoreWebView2NewWindowRequestedEventArgs args) in C:\Users\taparik\dev\maui\src\BlazorWebView\src\SharedSource\WebView2WebViewManager.cs:line 201
   at Microsoft.Web.WebView2.Core.CoreWebView2.OnNewWindowRequested(CoreWebView2NewWindowRequestedEventArgs args)
   at Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventHandler.Invoke(ICoreWebView2 source, ICoreWebView2NewWindowRequestedEventArgs args)
```

this was related to https://stackoverflow.com/a/2369171 and hence I'd hardcoded the iexplore path. Would you be open to `Process.Start` with a try/catch for a `Win32Exception` (as seen [here](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/csharp/general/start-internet-browser#complete-code-sample)). If we have `ErrorCode==-2147467259` we use `iexplorer`?

There's nothing special about my setup and I've tried both edge and chrome as the default `http`/`https` apps.

Reply:
I believe this may be related to https://github.com/dotnet/runtime/issues/15842.

Reply:
Okay yes, was related to `UseShellExecute`. Doing the following resolves the issue:

```csharp
using (var launchBrowser = new Process())
{
    launchBrowser.StartInfo.UseShellExecute = true;
    launchBrowser.StartInfo.FileName = uri.ToString();
    launchBrowser.Start();
}
```

Reply:
Hard-coding iexplore.exe is _never_ the right answer 😁